package/src/components/Channel/hooks/useCreateOwnCapabilitiesContext.ts

```diff
@@ -18,15 +18,27 @@ export const useCreateOwnCapabilitiesContext = <
   channel: Channel<StreamChatGenerics>;
   overrideCapabilities?: Partial<OwnCapabilitiesContextValue>;
 }) => {
+  const [own_capabilities, setOwnCapabilites] = useState(
+    JSON.stringify(channel.data?.own_capabilities as Array<string>),
+  );
   const overrideCapabilitiesStr = overrideCapabilities
     ? JSON.stringify(Object.values(overrideCapabilities))
     : null;
-  const ownCapabilitiesStr = channel?.data?.own_capabilities
-    ? JSON.stringify(Object.values(channel?.data?.own_capabilities as Array<string>))
-    : null;
+
+  useEffect(() => {
+    const listener = channel.on('capabilities.changed', (event) => {
+      if (event.own_capabilities) {
+        setOwnCapabilites(JSON.stringify(event.own_capabilities as Array<string>));
+      }
+    });
+
+    return () => {
+      listener.unsubscribe();
+    };
+  }, []);
 
```

Comment:
there is one issue here.. 

on every render we used to check `channel?.data?.own_capabilities` before and this is gone now.. 

the channel rerender is sometimes done using forceupdate.. 

this should be fixed!

Reply:
Fixed